cfme/utils/appliance/db.py

```diff
@@ -263,6 +263,44 @@ def loosen_pgssl(self, with_ssl=False):
         result = client.run_command("systemctl restart {scl}-postgresql".format(scl=scl))
         return result.rc
 
+    def create_db_lvm(self):
+        """
+        Set up a partition for the CFME DB to run on.
+
+        As a work-around for having to provide a separate disk to a CFME appliance
+        for the database, we instead partition the single disk we have and run
+        the DB on the new partition.
+
+        This requires that the appliance's disk has more space than CFME requires.
+        For example, on RHOS, downstream CFME uses 43GB on a disk but the flavor used
+        to deploy the template has a 75GB disk. Therefore, we have extra space which
+        we can partition.
+
+        Note that this is not the 'ideal' way of doing things and should
+        be a stop-gap measure until we are capabale of attaching additional disks to
+        an appliance via automation on all infra types.
+        """
+        client = self.ssh_client
+
+        self.logger.info("Creating new LVM for db using remaining space on /dev/vda...")
+
+        fstab_line = '/dev/mapper/dbvg-dblv $APPLIANCE_PG_MOUNT_POINT xfs defaults 0 0'
+        commands_to_run = [
+            'parted /dev/vda --script mkpart primary 43GB 48.5GB',
+            'pvcreate /dev/vda3',
+            'vgcreate dbvg /dev/vda3',
+            'lvcreate --yes -n dblv --size 5G dbvg',
+            'mkfs.xfs /dev/dbvg/dblv',
+            'echo -e "{}" >> /etc/fstab'.format(fstab_line),
+            'mount -a'
+        ]
+
+        for command in commands_to_run:
+            result = client.run_command(command)
+            if not result.success:
+                self.logger.warn("Command failed!")
```

Comment:
Log the command, RC, and output for future debugging

Reply:
run_command appears to log it. When I had the log line in there, it was printing twice..

The output is always logged in the self.logger.info below (whether success or fail). If its a failed RC we log that warning note.

Reply:
Yeah, see cfme/utils/ssh.py under "run_command":
```python
        logger.info("Running command %r", command)
```